pkg/operator2/operator.go

```diff
@@ -311,3 +431,34 @@ func getPrefixFilter() controller.Filter {
 		DeleteFunc: prefix,
 	}
 }
+
+func newHTTPSClient() (*http.Client, error) {
```

Comment:
Actually you do need to keep rebuilding since this will talk to a route.  As we discussed on slack, the TransportFor rest config logic is probably what you want.

Reply:
I keep the Transport and rebuild the client, not exactly sure whether I need to rebuild the Transport too. It seems to do the trick as is, though.